Merge experimental-transpile to master to fix #48

paulmedynski opened the pull request:
Merge the experimental-transpile branch back to master to address:

Use default compiler host with minimal tweaks #48

paulmedynski committed:
- Updated transpile to use the default compiler host provided by typescript, and only override the bare minimuim.
- This allows typescript to load dependencies and @types based on the target and/or lib configuration.
- Updated index to use typescript's line/col number mapping function to emit helpful errors.
- Added a barebones test app that can be used to manually verify compatibility with any version of typescript.

paulmedynski committed:
- Cleaned up the test app's README.
- Added some documentation on how to make compilation fail.

paulmedynski committed:
- Added ESLint config to ignore the nested node_modules/ in the test app, and to use ES2015 syntax.

paulmedynski committed:
- Fixed ESLint errors and setup a basic ESLint config to ensure that local testing matches the CI builds.

paulmedynski committed:
Removed unnecessary tslint.json file in the test app.

paulmedynski committed:
Added tslint config to brunch-config to fix problems in the test app.

joshdover commented:
Had issues when using this branch. The main one being that the imported modules weren't being included in the bundle. Each `require` would fail saying that the library wasn't present.

paulmedynski commented:
I think you'd need to tell Brunch to package any vendor and/or 3rd-party modules.  Are these modules that TypeScript needs to compile to JS, or are they JS modules sitting in a node_modules/ or vendor/ directory that you're just referencing from your TS code?

joshdover commented:
3rd party being imported in TS code. They were bundled before without this change (I believe). Either way, for my use-case I decided to go with Webpack. Thanks!

paulmedynski committed:
Added support for the lib compiler option that accepts the documented
--lib values and translates them into the filenames expected by TS
createProgram().

paulmedynski committed:
Removing compiler error suppression, and enabling full import resolution when compiling a file.
